staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go

```diff
@@ -197,12 +198,12 @@ func (c completedConfig) NewWithDelegate(delegationTarget genericapiserver.Deleg
 	s.GenericAPIServer.Handler.NonGoRestfulMux.Handle("/apis", apisHandler)
 	s.GenericAPIServer.Handler.NonGoRestfulMux.UnlistedHandle("/apis/", apisHandler)
 
-	apiserviceRegistrationController := NewAPIServiceRegistrationController(informerFactory.Apiregistration().InternalVersion().APIServices(), s)
+	apiserviceRegistrationController := NewAPIServiceRegistrationController(informerFactory.Apiregistration().V1().APIServices(), s)
 	availableController, err := statuscontrollers.NewAvailableConditionController(
-		informerFactory.Apiregistration().InternalVersion().APIServices(),
+		informerFactory.Apiregistration().V1().APIServices(),
 		c.GenericConfig.SharedInformerFactory.Core().V1().Services(),
 		c.GenericConfig.SharedInformerFactory.Core().V1().Endpoints(),
-		apiregistrationClient.Apiregistration(),
+		apiregistrationClient.ApiregistrationV1().(apiregistrationv1client.APIServicesGetter),
```

Comment:
why is this cast needed?

Reply:
oh nice catch, my IDE tooling complained so i did the cast i remember. it is indeed unnecessary.